src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs

```diff
@@ -916,6 +1033,137 @@ private MgmtLongRunningOperation AsMgmtOperation(LongRunningOperation operation)
             return mgmtOperation;
         }
 
+        private void WriteClientMethod(CodeWriter writer, ClientMethod clientMethod, ResourceOperation resourceOperation, BuildContext<MgmtOutputLibrary> context, bool async, string restClientName = "RestClient")
```

Comment:
Investigate if it is possible to reuse the `WriteClientMethod` in `MgmtClientBaseWriter`